front-end/src/components/Workspace.js

```diff
@@ -35,16 +35,27 @@ class Workspace extends React.Component {
     }
 
     /**
-     * serialize model and download as JSON file
+     * serialize model, POST to server, download response as JSON file
      */
-    save() {
+    async save() {
         const data = JSON.stringify(this.model.serialize());
-        const dataStr = "data:text/json;charset=utf-8," + encodeURIComponent(data)
-        const anchor = document.createElement("a")
-        anchor.href = dataStr;
-        anchor.download = "diagram.json";
-        anchor.click();
-        anchor.remove();
+        const resp = await fetch("/workflow/save", {
+            method: "POST",
+            headers: {"Content-Type": "application/json"},
+            body: data
+        });
+        if (resp.status !== 200) {
```

Comment:
You should add this check for 200 in the `FileUpload` function below as well.

Reply:
https://github.com/matthew-t-smith/visual-programming/pull/28/commits/b2404dd359d1d399613f9624d67ccf27492f5712